src/main/java/com/cloudbees/jenkins/GitHubWebHook.java

```diff
@@ -153,7 +157,10 @@ public static Jenkins getJenkinsInstance() throws IllegalStateException {
      * Other plugins may be interested in listening for these updates.
      *
      * @since 1.8
+     * @deprecated we do not think this API is required any more, if we are wrong, please raise a JIRA
      */
+    @Deprecated
+    @Restricted(NoExternalUse.class)
```

Comment:
From what I can see the only consumer of this API is the 1.x GitHub Branch Source. If there are other use cases for consumers remaining post SCM API 2.0.x updates then this would need to be augmented to pass through the origin.

I'd much rather kill off the code path, so by adding the `@Restricted` we will flush out any impls without breaking them until they update their dependency